DeepCat.js

```diff
@@ -209,7 +208,24 @@
 		newSearchTerms = DeepCat.computeResponses( responses, newSearchTerms );
 		newSearchTerms = DeepCat.computeErrors( errors, newSearchTerms );
 
-		substituteSearchRequest( newSearchTerms.join( ' ' ) );
+		logAndFinishRequest( newSearchTerms.join( ' ' ) );
+	}
+
+	function logAndFinishRequest( newSearchTermString ) {
+		var logRequestUrl = '//tools.wmflabs.org/catgraph-jsonp/logrequestlength?searchquerylength={0}';
+		return $.ajax( {
+			url: stringFormat( logRequestUrl, newSearchTermString.length ),
+			timeout: ajaxTimeout,
+			cache: false,
+			complete: function() {
+				finishDeepCatRequest( newSearchTermString );
+			}
+		} );
+	}
+
+	function finishDeepCatRequest( newSearchTermString ) {
+		substituteSearchRequest( newSearchTermString );
+		removeAjaxThrobber();
```

Comment:
When using the search with deepcat: as a user I would expect that clicking the submit button submits the form immediately and the page reloads as usual. 

This will and can not happen in the gadget before the above requests are finished.

That's why we are having the throbber. To show the user that there is something happening in the background and not leave him wondering why the form is not submitted yet. And that's the reason why I think the throbber should stop just right before the form gets finally submitted.